### PR TITLE
New Markers

### PR DESCRIPTION
**Pull request by fresqo**
Resolves Admin Markers

#### Changes proposed in this pull request: 
- Removal of server side Admin markers
- Use missioneventhandler for Map open/close
- Add admin choice of player/vehicle markers.
- small adjustments on current marker files for efficiency
- Admin has death markers like medics/police
- death markers are not looping setmarkerpos

CONCERNS (TO LOOK OVER AND FIX WHEN I GET MORE TIME)
- Every time dead player count changes, admin markers reset so they are live.
- Every time admin changes their player/vehicle marker selection the markers reset. 
- vehicle markers are poo yellow atm.. suggestions?
- Need to be able to see what is active at the time for admin markers (player/vehicle)

### CODE REVIEW

**Comment by Jawshy** [comment]
Travis has detected a lot of errors in those last few commits.

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 1]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
-	 	{
-		_deadCount = _deadCount + 1;	
-	 	};
-	}forEach allDeadMen;
-	_deadCount;
+    _deadCount = 0;
+    {
+        _name = _x getVariable "name";
+        _down = _x getVariable ["Revive",false];
+        if(!isNil "_name" && !_down) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 2]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
-			_marker setMarkerTextLocal format["%1", (_x select 3)];
-			_markers pushBack [_marker,(_x select 1),_x select 3,_x select 2];
-		};
-	} foreach _units;
+    _adminPlayerMarkerStatus = life_admin_playerMarkers;
+    _adminVehicleMarkerStatus = life_admin_vehicleMarkers;
+    private _units = [];
+    _markers = [];
+    {
+        if (life_admin_playerMarkers || life_admin_vehicleMarkers) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 3]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
-		};
-	} foreach _units;
+    _adminPlayerMarkerStatus = life_admin_playerMarkers;
+    _adminVehicleMarkerStatus = life_admin_vehicleMarkers;
+    private _units = [];
+    _markers = [];
+    {
+        if (life_admin_playerMarkers || life_admin_vehicleMarkers) then 
+        {
+            if (life_admin_vehicleMarkers) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 4]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
+    _adminVehicleMarkerStatus = life_admin_vehicleMarkers;
+    private _units = [];
+    _markers = [];
+    {
+        if (life_admin_playerMarkers || life_admin_vehicleMarkers) then 
+        {
+            if (life_admin_vehicleMarkers) then 
+            {
+                private _vehicles = vehicles;
+                if (count _vehicles > 0) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 5]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
+            };
+        };
+    } forEach playableUnits;
+
+    if (life_admin_playerMarkers && (_oldDeadCount > 0)) then 
+        {
+            {
+            _name = _x getVariable "name";
+            _down = _x getVariable ["Revive",false];
+            if(!isNil "_name" && !_down) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 6]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
+            _down = _x getVariable ["Revive",false];
+            if(!isNil "_name" && !_down) then 
+                {
+                    _units pushBack ["ColorRed",_x,"loc_Hospital",_x getVariable["realname",name _x]];
+                };
+            } foreach allDeadMen;
+        };
+
+        {
+        if((_x select 1) != player) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 7]
```diff
@@ -9,106 +9,107 @@ private _markers = [];
...
+    if ((call _fnc_deadCount) != _oldDeadCount) then 
+        {
+            {deleteMarkerLocal (_x select 0);} forEach _markers; 
+            _oldDeadCount = call _fnc_deadCount;
+            call _fnc_initMarkers;
+        };
+    {
+        private _marker = _x select 0;
+        private _unit = _x select 1;
+        if(!isNil "_unit" && !isNull _unit) then
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `if (`*

**Comment by fresqo** [comment]
yeah i know im still working on it atm.. will remove the WIP when im done.. sry

File: Altis_Life.Altis/core/medical/fn_medicMarkers.sqf [hunk 8]
```diff
@@ -5,57 +5,49 @@
...
-        if (!isNil "_name" && !_down) then {
-            _units pushBack _x;
-        };
-    } forEach allDeadMen;
+{
+if (side _x isEqualTo independent) then 
+	{
+		_units pushBack ["ColorIndependent",_x,"Mil_dot",_x getVariable["realname",name _x]];
+	};
+}forEach playableUnits;
```
Comment (Jawshy) [inline Altis_Life.Altis/core/medical/fn_medicMarkers.sqf]: `} foreach`*

File: Altis_Life.Altis/core/medical/fn_medicMarkers.sqf [hunk 9]
```diff
@@ -5,57 +5,49 @@
...
-            _markerss setMarkerTypeLocal "Mil_dot";
-            _markerss setMarkerTextLocal format ["%1", _x getVariable ["realname",name _x]];
+{
+	_name = _x getVariable "name";
+	_down = _x getVariable ["Revive",false];
+	if(!isNil "_name" && !_down) then 
+	{
+		_units pushBack ["ColorRed",_x,"loc_Hospital",_x getVariable["realname",name _x]];
+	};
+} foreach allDeadMen;
```
Comment (Jawshy) [inline Altis_Life.Altis/core/medical/fn_medicMarkers.sqf]: `forEach`*

File: Altis_Life.Altis/core/fn_setupEVH.sqf [hunk 10]
```diff
@@ -12,3 +12,30 @@ player addEventHandler["Take",{_this call life_fnc_onTakeItem}]; //Prevent peopl
...
+                [_this select 0] spawn life_fnc_adminMarkersActive;
+            };
+        } else {
+            {deleteMarkerLocal format["%1_marker",_x]} forEach allPlayers;
+            //{deleteMarkerLocal format["%1_marker",_x]} forEach allPlayers - entities "HeadlessClient_F";
+            if (count vehicles > 0) then 
+            {
+                {
+                    deleteMarkerLocal format["%1_marker",_x];
+                }forEach vehicles;
```
Comment (Jawshy) [inline Altis_Life.Altis/core/fn_setupEVH.sqf]: `} forEach`*

File: Altis_Life.Altis/core/cop/fn_copMarkers.sqf [hunk 11]
```diff
@@ -5,39 +5,49 @@
...
 
-sleep 0.5;
-if (visibleMap) then {
-    {if (side _x isEqualTo west) then {_cops pushBack _x;}} forEach playableUnits; //Fetch list of cops / blufor
+{
+if (side _x isEqualTo west) then 
+    {
+        _units pushBack ["ColorBLUFOR",_x,"Mil_dot",_x getVariable["realname",name _x]];
+    };
+}forEach playableUnits;
```
Comment (Jawshy) [inline Altis_Life.Altis/core/cop/fn_copMarkers.sqf]: `} forEach`*

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+        {
+        if ((_x select 1) != player) then 
+            {
+                private _marker = createMarkerLocal [format["%1_marker",(_x select 1)],visiblePosition (_x select 1)];
+                _marker setMarkerColorLocal (_x select 0);
+                _marker setMarkerTypeLocal (_x select 2);
+                _marker setMarkerTextLocal format["%1", (_x select 3)];
+                _markers pushBack [_marker,(_x select 1),_x select 3,_x select 2];
+            };
+        } foreach _units;
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: `forEach`*

**Comment by Jawshy** [comment]
[Altis_Life.Altis/core/admin/fn_adminMarkers.sqf](https://github.com/AsYetUntitled/Framework/blob/fresqo-patch-2/Altis_Life.Altis/core/admin/fn_adminMarkers.sqf): hints should be localised in the stringtable.

**Comment by fresqo** [comment]
will do strings now

File: Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf [hunk 13]
```diff
@@ -0,0 +1,115 @@
+/*
+	File: fn_adminMarkersActive.sqf
```
Comment (Jawshy) [inline Altis_Life.Altis/core/admin/fn_adminMarkersActive.sqf]: 4 spaces; not tabs. Even in comments.

File: Altis_Life.Altis/stringtable.xml [hunk 14]
```diff
@@ -779,6 +779,28 @@
...
             <Polish>Markery 'ON' graczy Włączone </Polish>
         </Key>
+        <Key ID="STR_ANOTF_VMDisabled">
+            <Original>Vehicle Markers Disabled.</Original>
+            <Czech>Značky vozidel zakázán.</Czech>
+            <French>Des marqueurs de véhicule en panne.</French>
+            <Spanish>Marcadores vehículo averiado.</Spanish>
+            <Italian>Marcatori veicoli disabilitati.</Italian>
+            <Russian>Маркеры транспортных средств отключены.</Russian>
+            <German>Fahrzeugmarkendeaktiviert.</German>
```
Comment (Jawshy) [inline Altis_Life.Altis/stringtable.xml]: @setoy, could you please check the accuracy of the German localisations?

File: Altis_Life.Altis/stringtable.xml [hunk 15]
```diff
@@ -779,6 +779,28 @@
...
+            <Polish>Markery pojazd wyłączony.</Polish>
+        </Key>
+        <Key ID="STR_ANOTF_VMEnabled">
+            <Original>Vehicle Markers Enabled.</Original>
+            <Czech>Omogućeno markeri vozila.</Czech>
+            <French>Des marqueurs de véhicules activés.</French>
+            <Spanish>Marcadores de vehículos habilitados.</Spanish>
+            <Italian>Marcatori dei veicoli abilitati.</Italian>
+            <Russian>включен маркеры транспортных средств.</Russian>
+            <German>Fahrzeugmarkern.</German>
```
Comment (Jawshy) [inline Altis_Life.Altis/stringtable.xml]: Ditto.

**Comment by fresqo** [comment]
awesome. i have no idea if languages other than English are okay, as i used google translate.. any help be great thanks.

**Comment by Jawshy** [comment]
@fresqo, they are and you did it spot on. Now we just need native/fluent speakers to verify the accuracy of the localisations.

**Comment by fresqo** [comment]
@baua1310 
Fahrzeugmarkierungen ausgeschaltet
Fahrzeugmarkerierungen angeschaltet
????? which one ??

**Comment by dbenson24** [comment]
@fresqo he commented them on line diffs, click the file name to see where.

Also just to clarify, are there multiple toggles now in the UK for Vehicles/Players/Dead People or just one all inclusive toggle?

**Comment by fresqo** [comment]
thanks @dbenson24 . no UI change. just an option for player or vehicle markers when markers is clicked. death markers show up if player markers are on.

**Comment by NiiRoZz** [comment]
It's a joke, all broken. If you active vehicle marker and vehicle dead, marker of vehicle will continue to alive. it's do anything like original;

**Comment by fresqo** [comment]
im still working on it... i have been testing

**Pull request closed without merging**